src/sentry/search/moresnuba/backend.py

```diff
@@ -43,8 +46,31 @@ class MoreSnubaSearchBackend(SnubaSearchBackend):
         "total": ["uniq", "events.issue"],
     }
 
-    # def initialize_group_queryset():
-    #     return Group.objects.all()
+    def build_group_queryset(
+        self, projects, environments, search_filters, retention_window_start, date_from, date_to
+    ):
+        # This is an override to see if we need to hit Postgres at all.
+        # If we don't, we just return None for a group_queryset and the result builder will not
+        # estimate the results or run pre/post filtering.
+
+        # We only need to hit postgres if search_filters contains certain fields.
+
+        self.search_postgres = False
```

Comment:
Please don't make this an attribute of the object. This is not accessed anywhere else.